src/components/CohortBuilder/SQONProvider.js

```diff
@@ -48,12 +89,31 @@ const SQONProvider = ({ children }) => {
     );
   const getActiveExecutableSqon = s => () =>
     resolveSyntheticSqon(s.state.sqons)(s.state.sqons[s.state.activeIndex]);
+  const onVirtualStudySelect = history => id => {
+    history.replace({
+      ...history.location,
+      search: stringify({ id }),
+    });
+  };
+
+  const setVirtualStudy = s => id => {
+    history.replace({
+      ...history.location,
+      search: stringify({ id }),
+    });
+    syncWithHistory(s);
+  };
 
   return (
-    <Component initialState={initialState} didMount={didMount} didUpdate={didUpdate}>
+    <Component
+      virtualStudyId={virtualStudyId}
```

Comment:
`virtualStudyId` need to be passed into react-component-component?

Reply:
so `didUpdate` receives it as props :)